test_winkernel produces wrong results with MIPS-32 (Big-endian) related tests

### Summary
test_winkernel does not produce the identical test results from the other existing test cases for all 10 MIPS-32 (Big-endian) related tests. Some of them print out different results and some others print out the below error message:

> ERROR: Failed to disasm given code!

### Impact
I think it is a minor issue because disassembling MIPS-32 (Big-endian) code from Windows drivers is extremely rare. While a root cause is unknown, it seems to affect only MIPS-32 (Big-endian) handling and no issue is seen on other targets from regression tests. 

### Step to reproduce
1. Compile the test_winkernel project on VS2015 or VS2013. It does not matter Debug or Release, Win32 or x64.
2. Copy compiled test_winkernel.sys to a test system and run it with DebugView (Kernel Capture enabled)
3. Save output to a file.
4. Compile test, test_detail, test_iter, and test_mips project on VS2015 or VS2013.
5. Run them on the same test system, save output to file(s).
6. Compare results in files.
7. All 10 tests named "* (Big-endian)" have different results (those of test_winkernel should be wrong). Affected tests and associated files are:

- test_detail (3 hits)
 - Platform: MIPS-32 (Big-endian)
 - Platform: MIPS-32R6 | Micro (Big-endian)
 - Platform: MIPS-32R6 (Big-endian)
- test_iter (1 hit)
 - Platform: MIPS-32 (Big-endian) 
- test_mips (3 hits)
 - Patform: MIPS-32 (Big-endian)
 - Platform: MIPS-32R6 | Micro (Big-endian)
 - Platform: MIPS-32R6 (Big-endian) 
- text (3 hits)
 - Platform: MIPS-32 (Big-endian)
 - Platform: MIPS-32R6 | Micro (Big-endian)
 - Platform: MIPS-32R6 (Big-endian)

### Reproduced test environment
commit: f0e528b079f01636d8ee308d6b2b56457bb748ba
platforms: Win10x64, Win10x86, Win7x64 (seems any platform)

### Note
- [both_test_results.zip](https://github.com/aquynh/capstone/files/250525/both_test_results.zip) contains outputs from both win_kernel and existing tests on Windows 10 x64.
- Using neither a checked build (a debug version of Windows) nor a DriverVerifier (a runtime driver error detector) identified the issue.

I have found that changing code in cs_open() solved the issue.
```
// from 
ud->big_endian = mode & CS_MODE_BIG_ENDIAN;
// to
ud->big_endian = (mode & CS_MODE_BIG_ENDIAN) != 0;
```
Code is setting 0 because the big_endian field is unsigned char and not real bool causing truncation of a result of the AND operation. 
```
struct cs_struct {
	...
	bool big_endian;

```
Here is a definition of bool in capstone_static_winkernel
```
// from platform.h

#if (_MSC_VER < 1800) || defined(_KERNEL_MODE)
// this system does not have stdbool.h
#ifndef __cplusplus
typedef unsigned char bool;
#define false 0
#define true 1
#endif    
```

As for a scope of this issue, I do think there is other instances of the same bug in the project because the warning C4244, which is triggered by the code, is not seen anywhere else, so the above fix should be fine. I will send a PR with test logs shortly.

--- a/cs.c
+++ b/cs.c
@@ -233,7 +233,7 @@ cs_err CAPSTONE_API cs_open(cs_arch arch, cs_mode mode, csh *handle)
 		ud->errnum = CS_ERR_OK;
 		ud->arch = arch;
 		ud->mode = mode;
-		ud->big_endian = mode & CS_MODE_BIG_ENDIAN;
+		ud->big_endian = (mode & CS_MODE_BIG_ENDIAN) != 0;
 		// by default, do not break instruction into details
 		ud->detail = CS_OPT_OFF;
 

